test/unit/network/ConnectionRegistryTest.js

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { expect } = require('chai');
+const sinon = require('sinon');
+const sandbox = sinon.createSandbox();
+
+const { ConnectionRegistryImpl } = require('../../../lib/network/ConnectionManager');
+const { Connection } = require('../../../lib/network/Connection');
+const { ConnectionStrategyConfigImpl, ReconnectMode } = require('../../../lib/config/ConnectionStrategyConfig');
+const { RoundRobinLB } = require('../../../lib/util/RoundRobinLB');
+const { UuidUtil } = require('../../../lib/util/UuidUtil');
+const { ClientOfflineError, IOError } = require('../../../lib/core/HazelcastError');
+
+describe('ConnectionRegistryTest', function () {
+
+    const connectionState = {
+        INITIAL: 0,
+        CONNECTED_TO_CLUSTER: 1,
+        INITIALIZED_ON_CLUSTER: 2
+    };
+
+    afterEach(function () {
+        sandbox.restore();
+    });
+
+    describe('getRandomConnection', function () {
+        it('should use load balancer in smart mode', function () {
```

Comment:
Ideally we should also test that the member UUID returned by the load balancer is actually used for the connection lookup. This could be done in a separate tests to keep them simple and self-contained.